### DISCUSSION TITLE
Trying to investigate a memory leak

### DISCUSSION THREAD

**Discussion by Hperigo** [gfx-rs/wgpu#5402] [Q&A]
Hello, I'm following learnOpenGl PBR section on IBL and I managed to get everything working but in the initial frames the app is allocating a huge amount of memory (around 90gb) and hanging for a couple of seconds and then the resources are freed 

I'm trying to make sense of what is happening and it looks like it's some internal resources created by wgpu, and the overall process for generating the environment maps are a bit complicated but I'll try to be brief 

1. Create a cubemap from equirectangular texture using a compute shader
2. generate diffuse convolution cube map using the prev generated cube map, using a render passes
3. generate specular convolution cube map using render passes
4. Generate a BRDF lut texture using compute shaders 

The weird part is that the app renders a couple of frames and then starts allocating a lot of resources stalling, so I'm not sure where this is coming from. 

Profiling with Xcode gives some insights, but I really don't know what to look for

The Specular convolution map has a bunch of these (wgpu internal) clear texture view generated and these stay around the lifetime of the program
<img width="2261" alt="Screenshot 2024-03-10 at 8 57 15 PM" src="https://github.com/gfx-rs/wgpu/assets/3333460/171ef02a-5c07-47a5-858f-75a72084ee08">

There are also multiple Texture2D allocated without any data, they are the same size of the display texture. so I imagine there's something wrong there..  (these get deleted once the program starts running agin)
<img width="2360" alt="Screenshot 2024-03-10 at 8 57 00 PM" src="https://github.com/gfx-rs/wgpu/assets/3333460/eb165462-3ccb-4257-aa41-41b28ffb0638">

here'e where I create the cubemap from equirectangular texture: 
https://github.com/Hperigo/pira-wgpu/blob/main/src/pipelines/sky/mod.rs#L161

and here is where I do the specular convolution
https://github.com/Hperigo/pira-wgpu/blob/main/src/pipelines/sky/mod.rs#L464

It's weird that the app draws a couple of frames (meaning, all the textures were created) and then it goes into some sort of data race and starts creating a bunch or resources... 

I wonder if anyone has any tips to try to track down this issue :/

**Comment by ltxitai** [comment]
The links you posted give 404 error - possibly the repository is set to private? 
If possible - it would be helpful to look at the code.

**Reply by Hperigo** [reply]
@ltxitai apologies for that, the repo is public now
